### PR TITLE
Fix options creation for @apollo/client >= 3.5

### LINKED ISSUE #133: Incompatibility/Bug with @apollo/client > 3.5.0
https://github.com/apollographql/apollo-client/commit/91945a1574aeb600b8faf6d099dbeabe1af866a1#diff-fb7fa652de6d84c08b87344a06a4d120b664c6e8bf491d16575794c755141477R356

A change was made in @apollo/client > 3.5 that means the `query` property in the options object will override the first argument and because rescript encodes `query: undefined` in that options object there's a runtime error. Technically I think the responsibility lies with apollo here, but humans probably wouldn't write `useQuery(documentNode, { query: undefined })` either.

The easiest way to fix this looks like replacing `query: None` with `query: Some(Operation.query)`, but would also be fixed by removing the `query` property entirely from the options type definitions. I'd be happy to do either just let me know what you'd prefer.

Also affects `useMutation`, `useLazyQuery`, etc

### PR DESCRIPTION
**Pull request by illusionalsagacity**
Resolves #133

### CODE REVIEW

File: src/@apollo/client/react/types/ApolloClient__React_Types.res [hunk 1]
```diff
@@ -730,44 +725,27 @@ module MutationHookOptions = {
     // export interface MutationHookOptions<TData = any, TVariables = OperationVariables> extends BaseMutationOptions<TData, TVariables> {
     //   mutation?: DocumentNode;
     // }
-    @ocaml.doc("
-     * We use deriving abstract here because option('jsVariables) is not typesafe,
-     * but still needs to be optional since a user has the option of providing
-     * the variables to the results of the hook rather than the hook itself
-     ")
```
Comment (illusionalsagacity) [inline src/@apollo/client/react/types/ApolloClient__React_Types.res:737]: 1. There's a few of these comments left over where the types aren't using `@deriving(abstract)` anymore, I can remove them if you think they aren't relevant anymore
2. `deriving(abstract)` seems to have the same effect as the `@obj` approach, figured I'd keep them all consistent but if there is a preference for one over the other I can change it.

**Comment by illusionalsagacity** [comment]
Smoke tested this on a branch of my employer's app, definitely fixes the immediate crash on startup.

File: src/@apollo/client/react/types/ApolloClient__React_Types.res (hunk 1 above)
Comment (jeddeloh) [reply src/@apollo/client/react/types/ApolloClient__React_Types.res:737]: I don't know that I prefer one over the other. I'd forgotten we use deriving actually. It does have the bonus of being able to convert in both directions, but in these cases, that's not actually happening. `@obj` seems fine to me. I think the general comment is still relevant. Maybe just switch deriving to `@obj`. I think there was an issue when nesting optionals or something that created an unexpected representation in some cases. Now I wish I'd written a better comment, though!

Thanks for going the extra mile for consistency!

File: src/@apollo/client/react/types/ApolloClient__React_Types.res [hunk 2]
```diff
@@ -22,28 +22,32 @@ module QueryHookOptions = {
     // export interface QueryHookOptions<TData = any, TVariables = OperationVariables> extends QueryFunctionOptions<TData, TVariables> {
     //     query?: DocumentNode;
     // }
-    type t<'jsData, 'jsVariables> = {
-      query: option<Graphql.documentNode>,
+    type t<'jsData, 'jsVariables>
+
+    @obj
```
Comment (jeddeloh) [inline src/@apollo/client/react/types/ApolloClient__React_Types.res:27]: Likewise, it might be worth a comment about why we are using `@obj`

**Review by jeddeloh — APPROVED** [review]
:shipit: Gonna wait to merge a bit in case you get to comments

**Comment by illusionalsagacity** [comment]
> :shipit: Gonna wait to merge a bit in case you get to comments

Yeah I can get to them in a bit

**Comment by jeddeloh** [comment]
Excellent! Thanks again! I also tested in /EXAMPLES and things look good. I checked that mutation options already use abstract/obj but subscription doesn't. I'm going to go ahead in a subsequent PR and convert subscription as well.

**Comment by jeddeloh** [comment]
Oh, ignore me. You already did it. Thanks!

**Pull request merged**